qa/tasks/cephfs/test_volume_client.py

```diff
@@ -686,15 +690,19 @@ def test_get_authorized_ids(self):
             vc.authorize(vp, "{guest_entity_1}", readonly=False)
             vc.authorize(vp, "{guest_entity_2}", readonly=True)
             auths = vc.get_authorized_ids(vp)
-            print auths
+            print(auths)
         """.format(
             group_id=group_id,
             volume_id=volume_id,
             guest_entity_1=guest_entity_1,
             guest_entity_2=guest_entity_2,
         )))
         # Check the list of authorized IDs and their access levels.
-        expected_result = [(u'guest1', u'rw'), (u'guest2', u'r')]
+        if self.py_version =='python3':
+            expected_result = [('guest1', 'rw'), ('guest2', 'r')]
+        else:
+            expected_result = [(u'guest1', u'rw'), (u'guest2', u'r')]
```

Comment:
Are you sure this needs to be done? The comparisons should still work.

Reply:
batrick wrote
> Are you sure this needs to be done? The comparisons should still work.
>
> rishabh-d-dave wrote:
>> I do remember running in to an error which led me to introduce these modifications but I could be wrong.

I used to and get the following error -
```
AssertionError: Element counts were not equal:
First has 2, Second has 6:  'u'
````

It's actually due to `assertItemsEqual()` and not python2/python3. Copying a snippet from the pdb session with breakpoint for the code above that I was using to recollect and investigate the error I was previously running into.

```
-> self.assertItemsEqual(str(expected_result), auths)
(Pdb) s
--Call--
> /usr/lib64/python2.7/unittest/case.py(902)assertItemsEqual()
-> def assertItemsEqual(self, expected_seq, actual_seq, msg=None):
(Pdb) n
> /usr/lib64/python2.7/unittest/case.py(915)assertItemsEqual()
-> first_seq, second_seq = list(expected_seq), list(actual_seq)
(Pdb) n
> /usr/lib64/python2.7/unittest/case.py(916)assertItemsEqual()
-> with warnings.catch_warnings():
(Pdb) p second_seq
['[', '(', 'u', "'", 'g', 'u', 'e', 's', 't', '2', "'", ',', ' ', 'u', "'", 'r', "'", ')', ',', ' ', '(', 'u', "'", 'g', 'u', 'e', 's', 't', '1', "'", ',', ' ', 'u', "'", 'r', 'w', "'", ')', ']']
(Pdb) p expected_seq
"[('guest1', 'rw'), ('guest2', 'r')]"
(Pdb) 
```

The tuples are converted into a list (of individual characters). As a result, the u-prefix for Unicode gets misinterpreted.